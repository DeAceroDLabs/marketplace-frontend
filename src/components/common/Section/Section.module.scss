@use "styles/typography";

.section {
  margin: auto;
  width: 90%;
}

.no-margin-title {
  font-size: typography.$header-m-size;
  font-weight: typography.$header-m-weight;
}

.margin-title {
  font-size: typography.$header-m-size;
  font-weight: typography.$header-m-weight;
  margin-bottom: 32px;
  margin-top: 40px;
}
