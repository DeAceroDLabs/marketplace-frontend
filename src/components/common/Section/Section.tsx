import styles from "./Section.module.scss";

interface SectionProps {
  title?: string | React.ReactNode;
  titleSize?: "small" | "medium" | "large";
  children?: React.ReactNode;
  variant?: "margin-title" | "no-margin-title";
  widthSize?: "medium" | "large";
}

const Section: React.FC<SectionProps> = ({
  title,
  titleSize ="medium",
  children,
  variant = "margin-title",
  widthSize = "large"
}) => {
  return (
    <div className={`${styles[`${widthSize}-container-width`]} ${styles.container}`}>
      <div className={`${styles[`${titleSize}-title`]} ${styles[variant]}`}>{title} </div>
      <div className={styles.children}> {children} </div>
    </div>
  );
};

export default Section;
