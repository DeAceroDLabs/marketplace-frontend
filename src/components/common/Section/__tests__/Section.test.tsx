import renderer from "react-test-renderer";
import Section from "../Section";

it("renders Section with no issue", () => {
  const component = renderer.create(
    <Section>
      <></>
    </Section>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
