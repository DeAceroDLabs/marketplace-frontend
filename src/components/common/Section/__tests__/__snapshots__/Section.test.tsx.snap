// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Section with no issue 1`] = `
<div
  className="large-container-width container"
>
  <div
    className="medium-title margin-title"
  >
     
  </div>
  <div
    className="children"
  >
     
     
  </div>
</div>
`;
